File: 0103-binary-tree-zigzag-level-order-traversal/0103-binary-tree-zigzag-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def zigzagLevelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:

        if root==None:
            return []
        
        ans =list()
        q = []
        q.append(root)
        move =1
        while len(q)>0:

            qSize = len(q)
            temp =list()
            for i in range(0,qSize):
                node = q.pop(0)
                if node.left!=None:
                    q.append(node.left)
                
                if node.right!=None:
                    q.append(node.right)
                temp.append(node.val)
            
            if move%2==0:
                temp.reverse()
                ans.append(temp)
            else:
                ans.append(temp)
            move+=1
        return ans

File: 0103-binary-tree-zigzag-level-order-traversal/0103-binary-tree-zigzag-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> zigzagLevelOrder(TreeNode* root) {
        if(root==NULL){
            return {};
        }
        queue<TreeNode*> q;
        int move=1;
        q.push(root);
        vector<vector<int>> ans;

        while(!q.empty()){
            int qSize = q.size();
            vector<int> temp;
            for(int i=0;i<qSize;i++){
                TreeNode* node = q.front();
                q.pop();
                if(node->left!=NULL){
                    q.push(node->left);
                }
                if(node->right!=NULL){
                    q.push(node->right);
                }
                temp.push_back(node->val);
            }
            if(move%2==0){
                reverse(temp.begin(),temp.end());
                ans.push_back(temp);
            }
            else{
                ans.push_back(temp);
            }
            move++;
        }
        return ans;
    }
};